File: app/src/main/java/com/dtu/capstone2/ereading/ui/newfeed/listnewfeed/ListNewFeedPagerAdapter.java
package com.dtu.capstone2.ereading.ui.newfeed.listnewfeed;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.dtu.capstone2.ereading.ui.model.ItemListNewFeedPager;

import java.util.List;

/**
 * Create by Nguyen Van Phuc on 4/8/19
 */
public class ListNewFeedPagerAdapter extends FragmentStatePagerAdapter {
    private List<ItemListNewFeedPager> mItemListNewFeedPagers;

    ListNewFeedPagerAdapter(FragmentManager fm, List<ItemListNewFeedPager> itemListNewFeedPagers) {
        super(fm);
        mItemListNewFeedPagers = itemListNewFeedPagers;
    }

    @Override
    public Fragment getItem(int i) {
        return mItemListNewFeedPagers.get(i).getFragment();
    }

    @Override
    public int getCount() {
        return mItemListNewFeedPagers.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mItemListNewFeedPagers.get(position).getTitleFragment();
    }
}

File: app/src/main/java/com/dtu/capstone2/ereading/ui/newfeed/listnewfeed/ListNewFeedPagerAdapter.kt
package com.dtu.capstone2.ereading.ui.newfeed.listnewfeed

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.dtu.capstone2.ereading.ui.model.ItemListNewFeedPager
import com.dtu.capstone2.ereading.ui.newfeed.listnewfeed.pagelistnewfeed.PageListNewFeedFragment

/**
 * Create by Nguyen Van Phuc on 4/8/19
 */
class ListNewFeedPagerAdapter(fm: FragmentManager, private val mItemListNewFeedPagers: List<ItemListNewFeedPager>) : FragmentStatePagerAdapter(fm) {

    override fun getItem(i: Int): Fragment {
        mItemListNewFeedPagers[i].also {
            return PageListNewFeedFragment.newInstant(it.urlEndPointRSS, it.titleFragment)
        }
    }

    override fun getCount(): Int = mItemListNewFeedPagers.size
}
